scripts/safe_template_linter.py

```diff
@@ -1464,244 +1852,137 @@ def check_python_file_is_safe(self, file_contents, results):
-                    # check if HTML string is appropriately wrapped
-                    for html_call in html_calls:
-                        if html_call.start_index < string.start_index < string.end_index < html_call.end_index:
-                            html_wrapped_strings.append(string)
-                            break
-            # loop through all unwrapped strings
-            for unsafe_string in set(html_strings) - set(html_wrapped_strings):
-                unsafe_string_expression = Expression(unsafe_string.start_index)
-                results.violations.append(ExpressionRuleViolation(
-                    Rules.python_wrap_html, unsafe_string_expression
-                ))
+        for match in re.finditer(r'\.display_name_with_default_escaped', file_contents):
+            expression = Expression(match.start(), match.end())
+            results.violations.append(ExpressionRuleViolation(
+                Rules.python_deprecated_display_name, expression
+            ))
 
-    def _check_interpolate_with_html(self, file_contents, start_index, end_index, results):
+    def _check_custom_escape(self, file_contents, results):
         """
-        Find interpolations with html that fall outside of any calls to HTML(),
-        Text(), and .format().
+        Checks for custom escaping calls, rather than using a standard escaping
+        method.
 
         Arguments:
             file_contents: The contents of the Python file
-            start_index: The index to start the search, or None if nothing to
-                search
-            end_index: The index to end the search, or None if the end of file
-                should be used.
             results: A list of results into which violations will be added.
 
         """
-        # used to find interpolation with HTML
-        pattern_interpolate_html_inner = r'(<.*%s|%s.*<|<.*{\w*}|{\w*}.*<)'
-        regex_interpolate_html = re.compile(r"""(".*{}.*"|'.*{}.*')""".format(
-            pattern_interpolate_html_inner, pattern_interpolate_html_inner
-        ))
-        if end_index is None:
-            interpolate_string_iter = regex_interpolate_html.finditer(file_contents, start_index)
-        else:
-            interpolate_string_iter = regex_interpolate_html.finditer(file_contents, start_index, end_index)
-        for match_html_string in interpolate_string_iter:
-            expression = Expression(match_html_string.start(), match_html_string.end())
+        for match in re.finditer("(<.*&lt;|&lt;.*<)", file_contents):
+            expression = Expression(match.start(), match.end())
             results.violations.append(ExpressionRuleViolation(
-                Rules.python_interpolate_html, expression
+                Rules.python_custom_escape, expression
             ))
 
 
 class MakoTemplateLinter(BaseLinter):
     """
     The linter for Mako template files.
     """
-    javaScriptLinter = JavaScriptLinter()
+    javascript_linter = JavaScriptLinter()
```

Comment:
Sure.  I'm not sure why they are out here, but probably from before __init__ and when I was struggling getting some Python class stuff working.